### ISSUE TITLE
[archlinux] USB icon does not show on Dock under archlinux

### ISSUE THREAD

**Issue by fasheng** [linuxdeepin/developer-center#51]
<a href="https://github.com/stevedude"><img src="https://avatars.githubusercontent.com/u/8483863?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by @stevedude**
_Thursday Feb 04, 2016 at 20:13 GMT_
_Originally opened as https://github.com/fasheng/arch-deepin/issues/238_

---

I am using Manjaro Deepin 15.12, 64-bit, 4.1 kernel. NVidia GTS 450 Graphics Card, AMD Phenom II Processor. There is an issue I am having:

Whenever I plug in a USB device (External Drive, Thumb Drive, etc) the USB icon does not appear on the Dock. I had it show up once, but not again, at all. I tried different USB devices and it does not work.

I can access the files and eject the USB drive from the file manager, but I would like to have access through the USB dock icon.

Thank you for a wonderful distribution.

**Comment by fasheng** [comment]
Looks udisks related, but I remember we did some work before, any idea? @jouyouyun

**Comment by jouyouyun** [comment]
Can you exec the command before plug in USB device? 
```shell
$ dbus-send --print-reply --dest=com.deepin.daemon.DiskMount /com/deepin/daemon/DiskMount org.freedesktop.DBus.Properties.Get string:"com.deepin.daemon.DiskMount" string:"DiskList"
$ dbus-monitor "path=/com/deepin/daemon/DiskMount"
```
Then plug in your USB device, wait for 1s, exec the first command in the above. Pasting all outputs, thanks!

**Comment by fasheng** [comment]
I found the ntfs/fat format usb dongle could not be auto-mounted after plugin, and the ext4 just works fine here.

**Comment by stevedude** [comment]
Sorry, I've been away. I did as you asked, I hope I did it correctly.

Output before I plugged-in my USB External Drive:

string "DiskList"
signal time=1459377874.228311 sender=:1.666 -> destination=(null destination) serial=810 path=/com/deepin/daemon/DiskMount; interface=org.freedesktop.DBus.Properties; member=PropertiesChanged
   string "com.deepin.daemon.DiskMount"
   array [
      dict entry(
         string "DiskList"
         variant             array [
               struct {
                  string "428 GB Volume"
                  string "native"
                  boolean false
                  boolean false
                  uint64 0
                  uint64 0
                  string "/dev/sda1"
                  string "6E82B93D82B90A9B"
                  string ""
                  string "drive-harddisk"
               }
               struct {
                  string "Ubuntu_Backup"
                  string "removable"
                  boolean true
                  boolean false
                  uint64 312823604
                  uint64 961302560
                  string "/dev/sdb1"
                  string "749b9de6-95d8-4f91-9214-fc9d49900bda"
                  string "file:///run/media/steve/Ubuntu_Backup"
                  string "drive-harddisk-usb"
               }
            ]
      )
   ]
   array [
   ]
method call time=1459377874.228830 sender=:1.663 -> destination=com.deepin.daemon.DiskMount serial=1188 path=/com/deepin/daemon/DiskMount; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.deepin.daemon.DiskMount"

After the command, my USB drive is recognized in the USB icon on the Deepin Dock. That never happened before.

Output after I plugged-in the USB External Drive:

string "DiskList"
signal time=1459378144.229992 sender=:1.666 -> destination=(null destination) serial=1043 path=/com/deepin/daemon/DiskMount; interface=org.freedesktop.DBus.Properties; member=PropertiesChanged
   string "com.deepin.daemon.DiskMount"
   array [
      dict entry(
         string "DiskList"
         variant             array [
               struct {
                  string "428 GB Volume"
                  string "native"
                  boolean false
                  boolean false
                  uint64 0
                  uint64 0
                  string "/dev/sda1"
                  string "6E82B93D82B90A9B"
                  string ""
                  string "drive-harddisk"
               }
               struct {
                  string "Ubuntu_Backup"
                  string "removable"
                  boolean true
                  boolean false
                  uint64 312823604
                  uint64 961302560
                  string "/dev/sdb1"
                  string "749b9de6-95d8-4f91-9214-fc9d49900bda"
                  string "file:///run/media/steve/Ubuntu_Backup"
                  string "drive-harddisk-usb"
               }
            ]
      )
   ]
   array [
   ]
method call time=1459378144.230394 sender=:1.663 -> destination=com.deepin.daemon.DiskMount serial=1447 path=/com/deepin/daemon/DiskMount; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.deepin.daemon.DiskMount"
   string "DiskList"
method call time=1459378144.231326 sender=:1.663 -> destination=com.deepin.daemon.DiskMount serial=1448 path=/com/deepin/daemon/DiskMount; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.deepin.daemon.DiskMount"
   string "DiskList"
method call time=1459378144.235597 sender=:1.663 -> destination=com.deepin.daemon.DiskMount serial=1449 path=/com/deepin/daemon/DiskMount; interface=org.freedesktop.DBus.Properties; member=Get
   string "com.deepin.daemon.DiskMount"
   string "DiskList"

**Comment by stevedude** [comment]
Also this particular External Drive "Ubuntu_Backup" should be formatted for Ext4 because it was my backup when I was running Ubuntu instead of Manjaro Deepin.

**Comment by fasheng** [comment]
Thanks, almost certainly it's a gir signal issue under gccgo now :(

**Comment by stevedude** [comment]
I see this is still open. I'm still running Manjaro Deepin (latest version) and I still cannot get my USB icons to work on the dock. Any news or progress?  Thank you.

**Comment by hualet** [comment]
@jouyouyun @felixonmargs

**Comment by felixonmars** [comment]
Looks like there are new places of failures for the -fno-plt -z,now problem I fixed last time.

**Comment by felixonmars** [comment]
Fixed in deepin-file-manager 1:4.2.2-2.

Upstream cr submitted: https://cr.deepin.io/#/c/25132/
